pandas/core/arrays/base.py

```diff
@@ -802,12 +809,20 @@ def fillna(
                 npvalues = self.astype(object)
                 func(npvalues, limit=limit, mask=mask)
                 new_values = self._from_sequence(npvalues, dtype=self.dtype)
+                if not copy:
+                    self[:] = new_values
             else:
                 # fill with value
-                new_values = self.copy()
+                if not copy:
+                    new_values = self[:]
+                else:
+                    new_values = self.copy()
                 new_values[mask] = value
         else:
-            new_values = self.copy()
+            if not copy:
+                new_values = self[:]
```

Comment:
Why do we need a shallow copy in these cases?

Reply:
i figured on principle to return a new object, doesnt make much difference

Reply:
How does this compare with other EA methods?

Reply:
i see a couple of `return self` and a couple of `return self[:]` in the base class methods. so not too consistent

Reply:
I don't care then, but we should make this consistent (not in this pr)

Reply:
im leaning towards consistently returning `self` as in the Block method we may want to determine whether a copy was made and `result is not values` is an easy way to do that.